packages/plugin-parser/src/Parser.ts

```diff
@@ -22,10 +23,21 @@ export class Parser<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<
     loadParsers(parsers: ParserConstructor[]): void {
         parsers = [...parsers].reverse();
         for (const ParserClass of parsers) {
-            const parsingEngine = this.engines[ParserClass.id];
-            if (parsingEngine) {
-                parsingEngine.register(ParserClass);
+            for (const id in this.engines) {
+                const parsingEngine = this.engines[id];
+                if (id === ParserClass.id) {
+                    parsingEngine.register(ParserClass);
+                } else {
+                    const fallbacks = parsingEngine.constructor.fallbacks;
+                    if (fallbacks && fallbacks.includes(ParserClass.id)) {
+                        parsingEngine.register(ParserClass);
+                    }
+                }
```

Comment:
Instead of this `if/else/if` construction, I suggest this:
```typescript
const supportedTypes = [id, ...parsingEngine.constructor.extends];
if (supportedTypes.includes(ParserClass.id) {
    parsingEngine.register(ParserClass);
}
```